src/usr/local/www/firewall_rules_edit.php

```diff
@@ -1090,6 +1084,17 @@ function build_if_list() {
 	return($iflist);
 }
 
+// Expands missing/blank/"all" icmptypes to a list of specific types. Doesn't check validity.
+function expand_icmptype($icmptype, $ipprotocol) {
+	if (isset($icmptype) && is_string($icmptype) && strlen($icmptype) > 0 && $icmptype != "any") {
```

Comment:
I just checked - if the source variable is unset (as icmptypes can conceivably be) the function is passed NULL as the value, not <undefined>. So this should be is_null() not isset(), thanks.